### DISCUSSION TITLE
Getting accurate pose and velocity data with Vicon

### DISCUSSION THREAD

**Discussion by eli-pratt** [bitcraze/discussions#1500] [Q&A]
We are working on implementing our own custom controller on the Crazyflie and need accurate position, orientation, linear velocity, and angular velocity data. We don't have any expansion decks on the drone but have been using a Vicon motion capture system to get position data. Using the Crazyswarm software, we have successfully streamed Vicon data to the drone. However, we have noticed some serious drift in the state data. For example, we had the drone sitting still on the floor, but the position data changed rapidly. The nonsense data seems to be coming from IMU drift over time. However, we have gotten the drone to hover and land successfully a few times, so the drone must be getting somewhat accurate data. 

Any tips on how we can accurately receive the data we need?

**Comment by knmcguire** [comment]
Since you are using the Crazyswarm2 package, it might be better to ask a question on their discussions: https://github.com/IMRCLab/crazyswarm2/discussions. They have a lot of mocap experience. Additionally you can also debug your system directly with the libmotioncapture library, to see if you are receiving good info in the first place: https://github.com/IMRCLab/libmotioncapture

**Reply by eli-pratt** [reply]
Okay, thank you! I have another question but it is a bit off topic so I will start a new thread. Thanks for the help!
